src/GlobalProvider.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { Todo } from './types/Todo';
+
+// function useLocalStorage<T>(
+//   key: string,
+//   defaultValue: T,
+// ): [T, React.Dispatch<React.SetStateAction<T>>] {
+//   const [value, setValue] = useState<T>(() => {
+//     const saved = localStorage.getItem(key);
+
+//     if (saved === null) {
+//       return defaultValue;
+//     }
+
+//     try {
+//       return JSON.parse(saved);
+//     } catch (e) {
+//       localStorage.removeItem(key);
+
+//       return defaultValue;
+//     }
+//   });
+
+//   React.useEffect(() => {
+//     localStorage.setItem(key, JSON.stringify(value));
+//   }, [key, value]);
+
+//   return [value, setValue];
+// }
+
+type RootState = {
+  todos: Todo[];
+};
+
+type Action =
+  | { type: 'getTodos' }
+  | { type: 'addTodo'; payload: Todo }
+  | { type: 'deleteTodo'; payload: { id: Todo['id'] } }
+  | { type: 'updateTodoStatus'; payload: { id: Todo['id']; complete: boolean } }
+  | { type: 'updateTodoTitle'; payload: { id: Todo['id']; title: string } };
```

Comment:
```suggestion
  | { type: 'updateTodoStatus'; payload: Pick<Todo, 'id' | 'complete'> }
  | { type: 'updateTodoTitle'; payload: Pick<Todo, 'id' | 'title'> };
```
Use utility types https://www.typescriptlang.org/docs/handbook/utility-types.html